crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -176,6 +176,48 @@ def create_proposal
     base
   end
 
+  def prepare_nodes_for_os_upgrade
+    all_nodes = NodeObject.all
+
+    all_nodes.each do |node|
+      next if node.admin? || node[:platform] == "windows"
+      node.set_state("os_upgrading")
+    end
+
+    # wait for the pxe_config to be updated, then reboot the nodes
+    discovery_dir = "#{NodeObject.admin_node[:provisioner][:root]}/discovery/"
+    pxecfg_subdir = "bios/pxelinux.cfg"
+
+    all_nodes.each do |node|
+      next if node.admin? || node["platform"] == "windows"
+
+      boot_ip_hex = node["crowbar"]["boot_ip_hex"]
+      node_arch = node["kernel"]["machine"] rescue "x86_64"
```

Comment:
@tboerger Do you think this needs to be addressed. If yes, what's a good way of fixing it?